src/chart/tree/TreeView.js

```diff
@@ -27,6 +27,85 @@ import View from '../../coord/View';
 import * as roamHelper from '../../component/helper/roamHelper';
 import RoamController from '../../component/helper/RoamController';
 import {onIrrelevantElement} from '../../component/helper/cursorHelper';
+import { __DEV__ } from '../../config';
+import {parsePercent} from '../../util/number';
+
+var TreeShape = graphic.extendShape({
+    shape: {
+        parentPoint: [],
+        childPoints: [],
+        orient: '',
+        forkPosition: ''
+    },
+
+    buildPath: function (ctx, shape) {
+        var ptMin = [Infinity, Infinity];
+        var ptMax = [-Infinity, -Infinity];
+        var points = shape.childPoints;
+        for (var i = 0; i < points.length; i++) {
+            var pt = points[i];
+            if (pt[0] < ptMin[0]) {
+                ptMin[0] = pt[0];
+            }
+            if (pt[0] > ptMax[0]) {
+                ptMax[0] = pt[0];
+            }
+            if (pt[1] < ptMin[1]) {
+                ptMin[1] = pt[1];
+            }
+            if (pt[1] > ptMax[1]) {
+                ptMax[1] = pt[1];
+            }
+        }
+        var parentPoint = shape.parentPoint;
+        var orient = shape.orient;
+        var location = parsePercent(shape.forkPosition, 1);
+        var midPoint = computeMidPoints(parentPoint, orient, ptMax, location);
+
+         ctx.moveTo(parentPoint[0], parentPoint[1]);
+         ctx.lineTo(midPoint[0], midPoint[1]);
+         if (orient === 'TB' || orient === 'BT') {
+             ctx.moveTo(ptMin[0], midPoint[1]);
+             ctx.lineTo(ptMax[0], midPoint[1]);
+         }
+         else if (orient === 'LR' || orient === 'RL') {
+             ctx.moveTo(midPoint[0], ptMin[1]);
+             ctx.lineTo(midPoint[0], ptMax[1]);
+         }
+         for (var i = 0; i < points.length; i++) {
+             var point = points[i];
+             ctx.moveTo(point[0], point[1]);
+             if (orient === 'TB' || orient === 'BT') {
+                 ctx.lineTo(point[0], midPoint[1]);
+             }
+             else if (orient === 'LR' || orient === 'RL') {
+                 ctx.lineTo(midPoint[0], point[1]);
+             }
+         }
+    }
```

Comment:
Top and bottom horizontal lines are better to be connected with the vertical line. 
```js
ctx.moveTo(); // top line end point
ctx.lineTo(); // top line connect point
ctx.lineTo(); // bottom line connect point
ctx.lineTo(); // bottom line end point
```

It will have a nicer corner than current if the line is thicker.
![image](https://user-images.githubusercontent.com/841551/70967923-d101d280-20d2-11ea-8a86-ccb35910d19c.png)